docs/proposals/rollout/README.md

```diff
@@ -0,0 +1,308 @@
+
+<!--
+What are the risks of this proposal, and how do we mitigate? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+- May further reduce the efficiency of distribution synchronization of workloads.
+
+## Architecture
+The primary approach involves using feedback from the StatusController on single-cluster workload statuses to derive the next rollout configurations. This ensures that throughout the Rollout process, rollout configurations such as `Partition`, `MaxUnavailable`, and `MaxSurge` maintain their original native meanings, and that the federated Rollout behavior aligns with single-cluster  practices.
+
+![architecture](rollout-mechanism-arch.png)
+
+## Changes required
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Code Changes required
+- New Methods in Resource Interpreter
+```golang
+package resourceinterpreter
+
+type ResourceInterpreter interface {
+	// Get & Set for Partition
+	GetPartition(object *unstructured.Unstructured) (*intstr.IntOrString, error)
+	RevisePartition(object *unstructured.Unstructured, partition *intstr.IntOrString) (*unstructured.Unstructured, error)
+
+	// Get & Set for MaxSurge
+	GetMaxSurge(object *unstructured.Unstructured) (*intstr.IntOrString, error)
+	ReviseMaxSurge(object *unstructured.Unstructured, maxSurge *intstr.IntOrString) (*unstructured.Unstructured, error)
+
+	// Get & Set for MaxUnavailble
+	GetMaxUnavailable(object *unstructured.Unstructured) (*intstr.IntOrString, err error)
+	ReviseMaxUnavailable(object *unstructured.Unstructured, maxUnavailable *intstr.IntOrString) (*unstructured.Unstructured, error)
+
+	RolloutInterperter
+}
+```
+
+- New Interpreter for Rollout
+```golang
+package rolloutinterpreter
+type RolloutInterperter interface {
+	ReviseRolloutConfigurations(resourceTemplate *unstructured.Unstructured, workloads *map[string]*unstructured.Unstructured, works map[string]*workv1alpha1.Work) (map[string]*unstructured.Unstructured, error)
+}
+
+type universalRolloutInterperter struct {
+}
+
+func (uri *universalRolloutInterperter) ReviseRolloutConfigurations(resourceTemplate *unstructured.Unstructured, workloads *map[string]*unstructured.Unstructured, works map[string]*workv1alpha1.Work) (map[string]*unstructured.Unstructured, error) {
+	modifiedWorkloads := make(map[string]*unstructured.Unstructured, len(workloads))
+	/*
+	   1. calculate partition for each workload if has partition interperter
+	   2. calculate maxUnavailble for each workload if has maxUnavailable interperter
+	   3. calculate maxSurge for each workload if has maxSurge interperter
+	   4. patch these calculated fields to each workload
```

Comment:
It's unclear to me yet if we need to parse anything from the status, like `updatedReplicas`, `readyReplicas`, etc?
If yet, brings another challenge that we need to `interpret` the status.